game-core/src/main/java/games/strategy/engine/lobby/client/ui/LobbyFrame.java

```diff
@@ -140,5 +140,6 @@ public void shutdown() {
               GameRunner.exitGameIfFinished();
             })
         .start();
+    return true;
```

Comment:
Hmm Methods returning the same objects regardless of execution path seem weird to me.
I'd actually move this true statement up one level and keep this method returning void, or alternatively have both frames implement an `QuitHandler` interface for consistency or something that will then be used.

Reply:
Implementing QuitHandler sounds interesting, I can take a look at doing it.

Reply:
I gave it a shot (didn't update the PR) and I'm not convinced it ends up cleaner.

At least, for TripleAFrame, if we implement the interface, then the non-Mac code that needs to attach the action to a JMenuItem would need to call it, which ends up looking like:

```
  public void shutdown() {
	handleQuitRequestWith(null, new QuitResponse() {
		@Override
		void cancelQuit() {
		}
		@Override
		void performQuit() {
		}
	});
  }
```

Which is messy. Or we make handleQuitRequestWith() call shutdown(), but then shutdown() needs to return a boolean as now.

So it doesn't end up cleaning that my current code. I could make LobbyFrame's shutdown() not return a boolean and handle it in the caller, but then it would be inconsistent.

I guess we could keep the version of setQuitHandler() that takes a Runnable (keep both signatures), and then LobbyFrame.shutdown() could be reverted to return void, while TripleAFrame would keep the code as-is.

What do you prefer @RoiEXLab ?

Reply:
Oh, that's not what I meant.
I was thinking about creating an interface like this
```java
interface QuitHandler {
  boolean shutdown();
}
```
which is then being implemented by both frames and instead of `BooleanSupplier` you pass the object directly

Reply:
Ah, I see. Done.